lib/fluent/registry.rb

```diff
@@ -60,31 +60,31 @@ def search(type)
       # search from additional plugin directories
       if @dir_search_prefix
         path = "#{@dir_search_prefix}#{type}"
-        files = @paths.map { |lp|
+        files = @paths.filter_map { |lp|
```

Comment:
Note: `filter_map` is available as of Ruby 2.7.
Recently we changed required Ruby version to 2.7, so it's no problem to use it: #4288